dask/bag/tests/test_bag.py

```diff
@@ -1122,7 +1122,7 @@ def test_to_delayed_optimize_graph():
     assert dict(d2.dask) == dict(x.dask)
     assert d.compute() == d2.compute()
 
-    [d] = b2.to_textfiles("foo.txt", compute=False)
+    [d] = b2.to_textfiles(tmpdir, compute=False)
```

Comment:
It looks like older versions of `fsspec` are having trouble with the `tmpdir` object. Converting it to a string, or using the `tmp_path` fixture, should resolve this issue:

```suggestion
    [d] = b2.to_textfiles(str(tmpdir), compute=False)
```

Reply:
It looks like in the other tests on the same file its used `str(tmpdir)` so we'll be consistent then.